File: java/rotational-cipher/src/main/java/RotationalCipher.java
class RotationalCipher {
    private int shiftKey;
    private static final String lettersLowerCase = "abcdefghijklmnopqrstuvwxyz";
    private static final String lettersUpperCase = "ABCDERGHIJKLMNOPQRSTUVWXYZ";

    RotationalCipher(int shiftKey) {
        this.shiftKey = shiftKey;
    }

    String rotate(String data) {
        return data.chars().map(this::rotateLetter).collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append).toString();
    }

    private int rotateLetter(int c) {
        if (Character.isAlphabetic(c)) {
            String letters = Character.isLowerCase(c) ? RotationalCipher.lettersLowerCase: RotationalCipher.lettersUpperCase;
            int index = letters.indexOf(c);
            return letters.codePointAt((index + this.shiftKey) % letters.length());
        }
        return c;
    }

}

File: csharp/rotational-cipher/RotationalCipher.cs
using System;
using System.Collections.Generic;
using System.Text;

public static class RotationalCipher
{
    private const string LowerCaseLetters = "abcdefghijklmnopqrstuvwxyz";
    private const string UpperCaseLetters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    public static string Rotate(string text, int shiftKey)
    {
        var cipher = new StringBuilder();
        foreach (var c in text)
        {
            if (char.IsLower(c))
            {
                cipher.Append(LowerCaseLetters[(LowerCaseLetters.IndexOf(c) + shiftKey) % 26]);
            }
            else if (char.IsUpper(c))
            {
                cipher.Append(UpperCaseLetters[(UpperCaseLetters.IndexOf(c) + shiftKey) % 26]);
            }
            else
            {
                cipher.Append(c);
            }
        }

        return cipher.ToString();
    }
}